src/main/java/com/FKFabian/medicalclinic/common/VisitValidator.java

```diff
@@ -0,0 +1,52 @@
+package com.FKFabian.medicalclinic.common;
+
+import com.FKFabian.medicalclinic.exceptions.ObjectAlreadyExistException;
+import com.FKFabian.medicalclinic.model.Visit;
+import com.FKFabian.medicalclinic.model.VisitCreateDto;
+import com.FKFabian.medicalclinic.repository.VisitRepository;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class VisitValidator {
+    private static VisitRepository visitRepository;
+
+    public static void checkIfVisitIsAvailable(VisitCreateDto visitCreateDto, List<Visit> visits) {
+        checkDate(visitCreateDto);
+        checkTime(visitCreateDto);
+        checkIfAnyVisitAlreadyExist(visitCreateDto, visits);
+    }
+
+    private static void checkDate(VisitCreateDto visitCreateDto) {
+        if (visitCreateDto.getStartingVisitDate().isBefore(LocalDateTime.now())) {
+            throw new IllegalArgumentException("Cannot create a visit for a past date.");
+        }
+    }
+
+    private static void checkTime(VisitCreateDto visitCreateDto) {
+        int minutes = visitCreateDto.getStartingVisitDate().getMinute();
+        if (minutes % 15 != 0) {
+            throw new IllegalArgumentException("Visit time must be in quarter-hour intervals.");
+        }
+    }
+
+    private static void checkIfAnyVisitAlreadyExist(VisitCreateDto visitCreateDto, List<Visit> visits) {
```

Comment:
Czy jest nam to potrzebne? W jednym momencie (w tym samym czasie) może mieć wizytę X lekarzy